src/dehydrated_devices.rs

```diff
@@ -21,6 +21,51 @@ impl From<dehydrated_devices::DehydratedDevices> for DehydratedDevices {
     }
 }
 
+/// Dehydrated device key
+#[wasm_bindgen]
+#[derive(Debug)]
+pub struct DehydratedDeviceKey {
+    pub(crate) inner: Uint8Array,
+}
+
+#[wasm_bindgen]
+impl DehydratedDeviceKey {
+    /// Generates a new random pickle key.
+    #[wasm_bindgen(js_name = "createRandomKey")]
+    pub fn create_random_key() -> Result<DehydratedDeviceKey, JsError> {
+        Ok(SdkDehydratedDeviceKey::new()?.into())
+    }
+
+    /// Generates a new random pickle key.
+    #[wasm_bindgen(js_name = "createKeyFromArray")]
+    pub fn create_key_from_array(array: Uint8Array) -> Result<DehydratedDeviceKey, JsError> {
+        Ok(SdkDehydratedDeviceKey::from_slice(array.to_vec().as_slice())?.into())
+    }
+
+    /// Convert the pickle key to a base 64 encoded string.
+    #[wasm_bindgen(js_name = "toBase64")]
+    pub fn to_base64(&self) -> JsString {
+        let binding = self.inner.to_vec();
+        let inner: &[u8; 32] = binding.as_slice().try_into().expect("Expected 32 byte array");
+
+        SdkDehydratedDeviceKey::from(inner).to_base64().into()
+    }
+}
+
+// Zero out on drop
+impl Drop for DehydratedDeviceKey {
+    fn drop(&mut self) {
+        self.inner.fill(0, 0, 32);
+    }
+}
+
+impl From<matrix_sdk_crypto::store::DehydratedDeviceKey> for DehydratedDeviceKey {
```

Comment:
it's a bit confusing that we use `matrix_sdk_crypto::store::DehydratedDeviceKey` and `SdkDehydratedDeviceKey` within the same file to refer to the same thing. I'd suggest using `SdkDehydratedDeviceKey` throughout, but please pick one and be consistent.

Reply:
Right tried to be consistent with other From<> impl. but will use the same.
Also there is some inconsistency in the was we name the import alias: sometimes it is `SdkThing` , `InnerThing` or `RustThing`. Let me try to clean that a bit

Reply:
oh I can see `OriginalThing` too